src/utils/config_utils/src/config_utils.c

```diff
@@ -101,11 +102,26 @@ static bool ADUC_AgentInfo_Init(ADUC_AgentInfo* agent, const JSON_Object* agent_
     }
 
     connection_type = json_object_get_string(connection_source, CONFIG_CONNECTION_TYPE);
-    connection_data = json_object_get_string(connection_source, CONFIG_CONNECTION_DATA);
+
+    // For v2.0, the connection data can be string or complex value (object)
```

Comment:
Would be simpler just to remove string support as not needed in v2 right?

Reply:
Still need it for testing V1 functionality (just in case)